src/ds3.c

```diff
@@ -428,6 +432,19 @@ void ds3_request_set_metadata(ds3_request* _request, const char* name, const cha
     g_free(prefixed_name);
 }
 
+void ds3_request_set_byte_range(ds3_request* _request, int64_t rangeStart, int64_t rangeEnd) {
+    char* range_value;
+    range_value = g_strdup_printf("bytes=%ld-%ld", rangeStart, rangeEnd);
```

Comment:
Right, but the Amazon API says it doesn't support that. I have no idea if
we follow what the Amazon API does or what HTTP spec says.

If we do want to support multiple ranges, the header bit wouldn't be too
hard, we could just check if the header exists, and if it does we append to
it. What would we do about the length and offset on the request?

On Fri, Dec 11, 2015 at 8:19 AM, Ryan Moore <notifications@github.com>
wrote:

> In src/ds3.c
> <https://github.com/SpectraLogic/ds3_c_sdk/pull/81#discussion_r47365343>:
>
> > @@ -428,6 +432,19 @@ void ds3_request_set_metadata(ds3_request* _request, const char* name, const cha
> >      g_free(prefixed_name);
> >  }
> >
> > +void ds3_request_set_byte_range(ds3_request* _request, int64_t rangeStart, int64_t rangeEnd) {
> > +    char* range_value;
> > +    range_value = g_strdup_printf("bytes=%ld-%ld", rangeStart, rangeEnd);
>
> The HTTP Range spec allows you to specify multiple ranges by separating
> them with commas. So If you had two ranges you could do Range:
> bytes=4-10,16-30
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SpectraLogic/ds3_c_sdk/pull/81/files#r47365343>.
>